### PR TITLE
Add fork and vfork stubs

### PR DESCRIPTION
**Pull request by sporksmith**

### CODE REVIEW

File: src/main/host/syscall/handler/clone.rs [hunk 1]
```diff
@@ -243,7 +237,25 @@ impl SyscallHandler {
...
         newtls: u64,
     ) -> Result<libc::pid_t, SyscallError> {
-        Self::clone_internal(ctx, flags_and_exit_signal, child_stack, ptid, ctid, newtls)
+        let raw_flags = flags_and_exit_signal as u32 & !0xff;
+        let raw_exit_signal = (flags_and_exit_signal as u32 & 0xff) as i32;
+
+        let Some(flags) = CloneFlags::from_bits(raw_flags as u64) else {
+            warn!("Couldn't parse clone flags: {raw_flags:x}");
+            return Err(Errno::EINVAL.into());
+        };
```
Comment (stevenengler) [inline src/main/host/syscall/handler/clone.rs:246]: I think this should be "debug"? This isn't an issue of shadow not implementing something correctly, it's an issue of the caller not giving a valid flag.

File: src/main/host/syscall/handler/clone.rs [hunk 2]
```diff
@@ -243,7 +237,25 @@ impl SyscallHandler {
...
+
+        let exit_signal = if raw_exit_signal == 0 {
+            None
+        } else {
+            let Ok(exit_signal) = Signal::try_from(raw_exit_signal) else {
+                warn!("Bad exit signal: {raw_exit_signal:?}");
+                return Err(Errno::EINVAL.into());
+            };
+            Some(exit_signal)
```
Comment (stevenengler) [inline src/main/host/syscall/handler/clone.rs:255]: "debug"?

File: src/main/host/syscall/handler/clone.rs [hunk 3]
```diff
@@ -262,45 +274,60 @@ impl SyscallHandler {
         }
         let args = ctx.objs.process.memory_borrow().read(args)?;
         trace!("clone3 args: {args:?}");
-        let Ok(flags) = i32::try_from(args.flags) else {
-            debug!("Couldn't safely truncate flags to 32 bits: {} ({:?})",
-                   args.flags, CloneFlags::from_bits(args.flags));
+        let Some(flags) = CloneFlags::from_bits(args.flags) else {
+            warn!("Couldn't parse clone flags: {:x}", args.flags);
```
Comment (stevenengler) [inline src/main/host/syscall/handler/clone.rs:278]: "debug"?

File: src/main/host/syscall/handler/clone.rs [hunk 4]
```diff
@@ -262,45 +274,60 @@ impl SyscallHandler {
...
+        )
+    }
+
+    #[log_syscall(/* rv */libc::pid_t)]
+    pub fn vfork(ctx: &mut SyscallContext) -> Result<libc::pid_t, SyscallError> {
+        // This should be the correct call to `clone_internal`, but `clone_internal`
+        // will currently return an error.
+        Self::clone_internal(
+            ctx,
+            CloneFlags::CLONE_VFORK,
```
Comment (stevenengler) [inline src/main/host/syscall/handler/clone.rs:322]: Shouldn't `vfork` also include `CLONE_VM`?

**Review by stevenengler — APPROVED** [review]

File: src/main/host/syscall/handler/clone.rs (hunk 1 above)
Comment (sporksmith) [reply src/main/host/syscall/handler/clone.rs]: done

File: src/main/host/syscall/handler/clone.rs (hunk 2 above)
Comment (sporksmith) [reply src/main/host/syscall/handler/clone.rs]: done

File: src/main/host/syscall/handler/clone.rs (hunk 3 above)
Comment (sporksmith) [reply src/main/host/syscall/handler/clone.rs]: done

File: src/main/host/syscall/handler/clone.rs (hunk 4 above)
Comment (sporksmith) [reply src/main/host/syscall/handler/clone.rs]: good catch!

**Pull request merged**